chain/sync.go

```diff
@@ -1038,6 +1043,8 @@ func (syncer *Syncer) syncMessagesAndCheckState(ctx context.Context, headers []*
 			return xerrors.Errorf("message processing failed: %w", err)
 		}
 
+		// Set current node sync height
+		stats.Record(ctx, metrics.ChainNodeHeight.M(int64(fts.TipSet().Height())))
```

Comment:
this is not getting you the nodes height, just the height of a tipset that this particular sync worker is churning on. To get the accepted height of the node, put this in chainstore takeHeaviestTipSet

Reply:
@magik6k Pointed me at doing a `headChangeNotifs` function so we would accurately trigger the head change in the metrics.

I noticed that these functions are not triggered until *after* the node has fully sync'd. Should I keep this method or inject the metric gathering into the `takeHeaviestTipSet` function itself? 

Here is the test function, which I tested and confirmed works (but only after fully sync'd):

```
	hcmetric := func(rev, app []*types.TipSet) error {
		ctx := context.Background()
		for _, r := range app {
			stats.Record(ctx, metrics.ChainHeight.M(int64(r.Height())))
		}
		return nil
	}

	cs.headChangeNotifs = append(cs.headChangeNotifs, hcnf, hcmetric)
```

I have this inside `NewChainStore` after the `hcnf` function is defined.